packages/universal-router-sdk/src/entities/protocols/pancakeswap.ts

```diff
@@ -0,0 +1,436 @@
+
+  for (const route of trade.routes) {
+    const { inputAmount, outputAmount, pools } = route
+    const amountIn: bigint = SmartRouter.maximumAmountIn(trade, options.slippageTolerance).quotient
+    const amountOut: bigint = SmartRouter.minimumAmountOut(trade, options.slippageTolerance).quotient
+
+    // flag for whether the trade is single hop or not
+    const singleHop = pools.length === 1
+
+    const recipient = routerMustCustody ? ROUTER_AS_RECIPIENT : validateAndParseAddress(options.recipient!)
+
+    const mixedRouteIsAllV3 = (r: Omit<BaseRoute, 'input' | 'output'>) => {
+      return r.pools.every(SmartRouter.isV3Pool)
+    }
+    const mixedRouteIsAllV2 = (r: Omit<BaseRoute, 'input' | 'output'>) => {
+      return r.pools.every(SmartRouter.isV2Pool)
+    }
+
+    const mixedRouteIsAllStable = (r: Omit<BaseRoute, 'input' | 'output'>) => {
+      return r.pools.every(SmartRouter.isStablePool)
+    }
+
+    // similar to encodeMixedRouteSwap but more simplified where we just continue
+    // as if its a regular v2 or v3 trade
+    if (singleHop) {
+      if (mixedRouteIsAllV3(route)) {
+        addV3Swap(
+          planner,
+          {
+            ...trade,
+            routes: [route],
+            inputAmount,
+            outputAmount,
+          },
+          options,
+          routerMustCustody,
+          payerIsUser,
+          performAggregatedSlippageCheck,
+        )
+      } else if (mixedRouteIsAllV2(route)) {
+        addV2Swap(
+          planner,
+          {
+            ...trade,
+            routes: [route],
+            inputAmount,
+            outputAmount,
+          },
+          options,
+          routerMustCustody,
+          payerIsUser,
+          performAggregatedSlippageCheck,
+        )
+      } else if (mixedRouteIsAllStable(route)) {
+        addStableSwap(
+          planner,
+          {
+            ...trade,
+            routes: [route],
+            inputAmount,
+            outputAmount,
+          },
+          options,
+          routerMustCustody,
+          performAggregatedSlippageCheck,
+          payerIsUser,
+        )
+      } else {
+        throw new Error('Unsupported route to encode')
+      }
+    } else {
+      const sections = SmartRouter.partitionMixedRouteByProtocol(route)
+
+      const isLastSectionInRoute = (i: number) => {
+        return i === sections.length - 1
+      }
+
+      let outputToken
+      let inputToken = inputAmount.currency.wrapped
+
+      for (let i = 0; i < sections.length; i++) {
+        const section = sections[i]
+        /// Now, we get output of this section
+        outputToken = SmartRouter.getOutputOfPools(section, inputToken)
+        const newRoute = SmartRouter.buildBaseRoute([...section], inputToken, outputToken)
+        inputToken = outputToken.wrapped
+
+        const lastSectionInRoute = isLastSectionInRoute(i)
+        const recipientAddress = isLastSectionInRoute(i) ? recipient : ROUTER_AS_RECIPIENT
```

Comment:
let's use the variable `lastSectionInRoute`

Reply:
check the protocol of next section here. If the protocol of next section is v2, set the recipient to the first pair. Otherwise use router as recipient.

Reply:
yeah dont know why i doing that and then not using it haha